coder_sniffer/Drupal/Sniffs/Commenting/TodoCommentSniff.php

```diff
@@ -0,0 +1,157 @@
+        } else if ($tokens[$stackPtr]['code'] === T_DOC_COMMENT_TAG) {
+            // Document comment tag (i.e. comments that begin with "@").
+            // Determine if this is related at all and build the full comment line
+            // from the various segments that the line is parsed into.
+            $expression = '/^@to/i';
+            $comment    = $tokens[$stackPtr]['content'];
+            if ((bool) preg_match($expression, $comment) === true) {
+                if ($this->debug === true) {
+                    echo "Attempting to build comment\n";
+                }
+
+                $index = ($stackPtr + 1);
+                while ($tokens[$index]['line'] === $tokens[$stackPtr]['line']) {
+                    $comment .= $tokens[$index]['content'];
+                    $index++;
+                }
+
+                if ($this->debug === true) {
+                    echo "Result comment = $comment\n";
+                }
+
+                $this->checkTodoFormat($phpcsFile, $stackPtr, $comment);
+            }//end if
+        }//end if
+
+    }//end process()
+
+
+    /**
+     * Checks a comment string for the correct syntax.
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param int                         $stackPtr  The position of the current token
+     *                                               in the stack passed in $tokens.
+     * @param string                      $comment   The comment text.
+     *
+     * @return void
+     */
+    private function checkTodoFormat(File $phpcsFile, $stackPtr, string $comment)
+    {
+        if ($this->debug === true) {
+            echo "Checking \$comment = '$comment'\n";
+        }
+
+        $expression = '/(?x)   # Set free-space mode to allow this commenting
+            ^(\/|\s)*          # At the start optionally match any forward slashes and spaces
+            (?i)               # set case-insensitive mode
+            (?=(               # start a positive non-consuming look-ahead to find all possible todos
+              @+to(-|\s|)+do   # if one or more @ allow spaces and - between the to and do
+              |                # or
+              to(-)*do         # if no @ then only accept todo or to-do or to--do, etc, no spaces
+            ))
+            (?-i)              # Reset to case-sensitive
+            (?!                # Start another non-consuming look-ahead, this time negative
+              @todo\s          # It has to match lower-case @todo followed by one space
+              (?!-|:)\S        # and then any non-space except - or :
+            )/m';
+
+        if ((bool) preg_match($expression, $comment) === true) {
+            if ($this->debug === true) {
+                echo "Failed regex - give message\n";
+            }
+
+            $comment = trim($comment, " /\r\n");
+            $phpcsFile->addWarning("'%s' should match the format '@todo Some task'", $stackPtr, 'TodoFormat', [$comment]);
```

Comment:
Comments should end with a dot, also todo comments. So the example should be something like `@todo Fix problem X here.`

Reply:
It would be really cool to have a fixer here, that would help many people auto-fix this standard (also in Drupal core). I think it should be possible, we should have all info in the regex and could try a preg_replace().

Not a blocker for this pull request, we can merge this as first step.

Reply:
Yes, doc comments not ending with a period is checked for. But currently @ tags are not considered within that group, so no warning is given if they have no period at the end. I agree that if they should end in a dot then this example should include the dot. But then we should also include @ tags in the general sniff for no ending period. @see should not be included, we discussed that on the deprecation sniffs, the url should not end in a dot. Maybe this is for another issue?

Regarding a fixer, yes it should be possible. @ adamzimmermann did suggest that, but we wanted to wait until we knew you were interested in this sniff

Reply:
I see, very considerate of you! Yep, I absolutely support this sniff.

Also yep to the ending dot discussion, should be a separate issue.

Reply:
> Also yep to the ending dot discussion, should be a separate issue.

I have raised https://www.drupal.org/project/coder/issues/3183156 to check for doc tags not ending with a period.